src/ir/wrap.jl

```diff
@@ -66,15 +66,30 @@ function IRCode(ir::IR)
   end
 end
 
+if VERSION >= v"1.6.0-DEV.272"
+    isgotoifnot(@nospecialize(ex)) = (@assert !isexpr(ex, :gotoifnot); ex isa Core.GotoIfNot)
+    destruct_gotoifnot(@nospecialize(ex)) = Any[ex.cond, ex.dest]
+
+    isreturn(@nospecialize(ex)) = (@assert !isexpr(ex, :return); ex isa ReturnNode)
+    retval(@nospecialize(ex)) = ex.val
+else
+    isgotoifnot(@nospecialize(ex)) = isexpr(ex, :gotoifnot)
+    destruct_gotoifnot(@nospecialize(ex)) = ex.args
+
+    isreturn(@nospecialize(ex)) = isexpr(ex, :return)
+    retval(@nospecialize(ex)) = ex.args[1]
+end
+
 function blockstarts(ci::CodeInfo)
   bs = Int[]
   terminator = false
   for i = 1:length(ci.code)
     ex = ci.code[i]
-    if isexpr(ex, :gotoifnot)
-      push!(bs, ex.args[2])
+    if isgotoifnot(ex)
+      _, dest = destruct_gotoifnot(ex)
+      push!(bs, dest)
       terminator = true
-    elseif isexpr(ex, GotoNode, :return)
+    elseif ex isa GotoNode || isreturn(ex)
```

Comment:
Shouldn't this be `&&`?

Reply:
No, I am pretty sure this is correct: https://github.com/FluxML/MacroTools.jl/blob/a03742057f388899833c9000ebf56dbd9616a6d5/src/utils.jl#L39. `ex` can only be either a `GotoNode` or a `ReturnNode`, not both.

Reply:
cool